src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1221,6 +1493,13 @@ public override BoundNode VisitLocal(BoundLocal node)
             var local = node.LocalSymbol;
             int slot = GetOrCreateSlot(local);
             var type = GetDeclaredLocalResult(local);
+
+            if (!node.Type.Equals(type.TypeSymbol, TypeCompareKind.ConsiderEverything | TypeCompareKind.IgnoreNullableModifiersForReferenceTypes | TypeCompareKind.IgnoreDynamicAndTupleNames))
```

Comment:
>if (!node.Type.Equals(type.TypeSymbol [](start = 12, length = 37)

I didn't understand this case. What's a scenario where they could be different?

Reply:
In error scenarios, the type of the `LocalSymbol` and the `BoundLocal` will differ.

---
In reply to: [265698571](https://github.com/dotnet/roslyn/pull/34017#discussion_r265698571) [](ancestors = 265698571)

Reply:
I understand that's not related to this PR, but that seems very strange... :-S

---
In reply to: [265700285](https://github.com/dotnet/roslyn/pull/34017#discussion_r265700285) [](ancestors = 265700285,265698571)

Reply:
That was my reaction when I saw that this was possible.

---
In reply to: [265761468](https://github.com/dotnet/roslyn/pull/34017#discussion_r265761468) [](ancestors = 265761468,265700285,265698571)

Reply:
I wonder if we should file a follow-up issue to investigate this. @cston what do you think?

---
In reply to: [265771984](https://github.com/dotnet/roslyn/pull/34017#discussion_r265771984) [](ancestors = 265771984,265761468,265700285,265698571)

Reply:
@333fred, what is an example where we hit this? Isn't it a `var` local declaration where the local is used in the initializer? Regardless, we should probably log a bug to have the type of the reference match the declared type.

---
In reply to: [265788526](https://github.com/dotnet/roslyn/pull/34017#discussion_r265788526) [](ancestors = 265788526,265771984,265761468,265700285,265698571)

Reply:
@cston this appears in error scenarios when a variable with inferred type is used before initialization. For example, `InverLocalType_UsedBeforeDeclaration`

---
In reply to: [265790001](https://github.com/dotnet/roslyn/pull/34017#discussion_r265790001) [](ancestors = 265790001,265788526,265771984,265761468,265700285,265698571)

Reply:
Filed [https://github.com/dotnet/roslyn/issues/34158](https://github.com/dotnet/roslyn/issues/34158) to follow up, and I'll leave a comment.

---
In reply to: [266089099](https://github.com/dotnet/roslyn/pull/34017#discussion_r266089099) [](ancestors = 266089099,265790001,265788526,265771984,265761468,265700285,265698571)